### DISCUSSION TITLE
How to customize the encoding name case?

### DISCUSSION THREAD

**Discussion by pedoc** [dotnet/runtime#84898] [Q&A]
When I use code like below for xml serialization,
```charp
using (var streamWriter = new StreamWriter(output, new UTF8Encoding())
    {
        serializer.Serialize(streamWriter, obj);
    }
```
will output
```xml
<?xml version="1.0" encoding="utf-8"?>
<output></output>
```

Hi everyone,I can integrate from the UTF8Encoding type and rewrite some members to make the output encoding name uppercase, the code is as follows
```csahrp
class CustomUTF8Encoding : UTF8Encoding
        {
            public override string HeaderName => base.HeaderName.ToUpper();
            public override string WebName => base.WebName.ToUpper();
            public override string BodyName => base.BodyName.ToUpper();
        }
using (var streamWriter = new StreamWriter(output, new CustomUTF8Encoding ())
    {
        serializer.Serialize(streamWriter, obj);
    }
```
will output
```xml
<?xml version="1.0" encoding="UTF-8"?>
<output></output>
```

The problem is that I currently use GB2312 encoding, and its real type is [DBCSCodePageEncoding. cs](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Encoding.CodePages/src/System/Text/DBCSCodePageEncoding.cs), which is an inner class, I cannot inherit and override the name field.

So what should I meet to complete this task? 
Thanks in advance

**Comment by colejohnson66** [comment]
Why does it need to be uppercase in the first place? Someone correct me if I'm wrong, but doesn't the XML spec require it be case-insensitive when checked?

**Comment by pedoc** [comment]
> but doesn't the XML spec require it be case-insensitive when checked?

Yes, but not all implementations are fully compliant. Some programs use concatenated strings to construct xml content, and do not use standard xml serialization (some java programs).

At the same time, they parse xml and directly manipulate strings, so they don't follow the standard.
